go/libraries/doltcore/sqle/dolt_log_table_function.go

```diff
@@ -623,78 +624,84 @@ func (ltf *LogTableFunction) NewDotDotLogTableFunctionRowIter(ctx *sql.Context,
 // Next retrieves the next row. It will return io.EOF if it's the last row.
 // After retrieving the last row, Close will be automatically closed.
 func (itr *logTableFunctionRowIter) Next(ctx *sql.Context) (sql.Row, error) {
-	var hashToLog hash.Hash
-	var cmToLog *doltdb.Commit
+	var h hash.Hash
+	var cm *doltdb.Commit
+	var err error
 	for {
-		h, cm, err := itr.child.Next(ctx)
+		h, cm, err = itr.child.Next(ctx)
 		if err != nil {
 			return nil, err
 		}
 
 		if itr.tableNames != nil {
-			if itr.prevCommit == nil {
-				itr.prevCommit = cm
-				itr.prevHash = h
+			if cm.NumParents() == 0 {
```

Comment:
Why continuing here? In the case where you weren't filtering on tables, that commit would be returned. I believe this will result in skipping a commit, but I may be reading this wrong.